ydb/core/wrappers/fs_storage.cpp

```diff
@@ -0,0 +1,494 @@
+        }
+        ReplyAdapter.Reply(sender, std::move(response));
+    }
+
+public:
+    TFsOperationActor(const TString& basePath, bool verbose, const TReplyAdapterContainer& replyAdapter)
+        : BasePath(basePath)
+        , Verbose(verbose)
+        , ReplyAdapter(replyAdapter)
+    {
+        FS_LOG(Verbose, "TFsOperationActor created: BasePath# " << BasePath);
+    }
+
+    ~TFsOperationActor() {
+        CleanupActiveSessions();
+    }
+
+    void Bootstrap() {
+        FS_LOG(Verbose, "TFsOperationActor Bootstrap called");
+        Become(&TThis::StateWork);
+    }
+
+    void PassAway() override {
+        CleanupActiveSessions();
+        NActors::TActorBootstrapped<TFsOperationActor>::PassAway();
+    }
+
+private:
+    void CleanupActiveSessions() {
+        if (!ActiveUploads.empty()) {
+            FS_LOG(Verbose, "TFsOperationActor: cleaning up " << ActiveUploads.size() << " active MPU sessions");
+            for (auto& [uploadId, session] : ActiveUploads) {
+                try {
+                    session->File.Flush();
+                    session->File.Close();
+                    FS_LOG(Verbose, "TFsOperationActor: closed MPU session: uploadId# " << uploadId);
+                } catch (const std::exception& ex) {
+                    LOG_ERROR_S(*TlsActivationContext, NKikimrServices::FS_WRAPPER,
+                        "Failed to close MPU session: uploadId# " << uploadId << ", error# " << ex.what());
+                }
+            }
+            ActiveUploads.clear();
+        }
+    }
+
+public:
+
+    STATEFN(StateWork) {
+        FS_LOG(Verbose, "TFsOperationActor StateWork received event type# " << ev->GetTypeRewrite());
+        switch (ev->GetTypeRewrite()) {
+            hFunc(TEvPutObjectRequest, Handle);
+            hFunc(TEvGetObjectRequest, Handle);
+            hFunc(TEvHeadObjectRequest, Handle);
+            hFunc(TEvDeleteObjectRequest, Handle);
+            hFunc(TEvCheckObjectExistsRequest, Handle);
+            hFunc(TEvListObjectsRequest, Handle);
+            hFunc(TEvDeleteObjectsRequest, Handle);
+            hFunc(TEvCreateMultipartUploadRequest, Handle);
+            hFunc(TEvUploadPartRequest, Handle);
+            hFunc(TEvCompleteMultipartUploadRequest, Handle);
+            hFunc(TEvAbortMultipartUploadRequest, Handle);
+            hFunc(TEvUploadPartCopyRequest, Handle);
+            sFunc(NActors::TEvents::TEvPoison, PassAway);
+            default:
+                LOG_WARN_S(*TlsActivationContext, NKikimrServices::FS_WRAPPER,
+                    "TFsOperationActor StateWork received unknown event type# " << ev->GetTypeRewrite());
+        }
+    }
+
+    void Handle(TEvPutObjectRequest::TPtr& ev) {
+        const auto& request = ev->Get()->GetRequest();
+        const auto& body = ev->Get()->Body;
+        const TString key = TString(request.GetKey().data(), request.GetKey().size());
+
+        FS_LOG(Verbose, "FS PutObject: key# " << key << ", size# " << body.size());
```

Comment:
Рекомендую такое форматирование:
```suggestion
        FS_LOG(Verbose, "PutObject"
            << ": key# " << key
            << ", size# " << body.size());
```